src/resources/Element/Token.ts

```diff
@@ -0,0 +1,207 @@
+import Resource from '../../resource';
+import Configuration from '../../configuration';
+
+export const UserEventType = {
+  AUTH_INTRO_OPEN: 'AUTH_INTRO_OPEN',
+  AUTH_INTRO_CONTINUE: 'AUTH_INTRO_CONTINUE',
+  AUTH_INTRO_CLOSE: 'AUTH_INTRO_CLOSE',
+
+  AUTH_NAME_OPEN: 'AUTH_NAME_OPEN',
+  AUTH_NAME_CONTINUE: 'AUTH_NAME_CONTINUE',
+  AUTH_NAME_CLOSE: 'AUTH_NAME_CLOSE',
+
+  AUTH_PHONE_OPEN: 'AUTH_PHONE_OPEN',
+  AUTH_PHONE_CONTINUE: 'AUTH_PHONE_CONTINUE',
+  AUTH_PHONE_CLOSE: 'AUTH_PHONE_CLOSE',
+
+  AUTH_PHONE_VERIFY_OPEN: 'AUTH_PHONE_VERIFY_OPEN',
+  AUTH_PHONE_VERIFY_SUBMIT: 'AUTH_PHONE_VERIFY_SUBMIT',
+  AUTH_PHONE_VERIFY_CLOSE: 'AUTH_PHONE_VERIFY_CLOSE',
+
+  AUTH_DOB_OPEN: 'AUTH_DOB_OPEN',
+  AUTH_DOB_CONTINUE: 'AUTH_DOB_CONTINUE',
+  AUTH_DOB_CLOSE: 'AUTH_DOB_CLOSE',
+
+  AUTH_ADDRESS_OPEN: 'AUTH_ADDRESS_OPEN',
+  AUTH_ADDRESS_CONTINUE: 'AUTH_ADDRESS_CONTINUE',
+  AUTH_ADDRESS_CLOSE: 'AUTH_ADDRESS_CLOSE',
+
+  AUTH_INCORRECT_INFO_OPEN: 'AUTH_INCORRECT_INFO_OPEN',
+  AUTH_INCORRECT_INFO_TRY_AGAIN: 'AUTH_INCORRECT_INFO_TRY_AGAIN',
+
+  AUTH_INVALID_INFO_OPEN: 'AUTH_INVALID_INFO_OPEN',
+  AUTH_INVALID_INFO_EXIT: 'AUTH_INVALID_INFO_EXIT',
+
+  AUTH_SECQ_OPEN: 'AUTH_SECQ_OPEN',
+  AUTH_SECQ_CONTINUE: 'AUTH_SECQ_CONTINUE',
+  AUTH_SECQ_CLOSE: 'AUTH_SECQ_CLOSE',
+
+  AUTH_SECQ_INCORRECT_OPEN: 'AUTH_SECQ_INCORRECT_OPEN',
+  AUTH_SECQ_INCORRECT_TRY_AGAIN: 'AUTH_SECQ_INCORRECT_TRY_AGAIN',
+  AUTH_SECQ_INCORRECT_CLOSE: 'AUTH_SECQ_INCORRECT_CLOSE',
+
+  AUTH_CONSENT_OPEN: 'AUTH_CONSENT_OPEN',
+  AUTH_CONSENT_CONTINUE: 'AUTH_CONSENT_CONTINUE',
+  AUTH_CONSENT_CLOSE: 'AUTH_CONSENT_CLOSE',
+
+  AUTH_SUCCESS_OPEN: 'AUTH_SUCCESS_OPEN',
+  AUTH_SUCCESS_CONTINUE: 'AUTH_SUCCESS_CONTINUE',
+  AUTH_FAILURE_OPEN: 'AUTH_FAILURE_OPEN',
+  AUTH_FAILURE_CONTINUE: 'AUTH_FAILURE_CONTINUE',
+};
+
+export type TUserEventType = keyof typeof UserEventType;
+
+export interface IElementUserEvent {
+  type: TUserEventType,
+  timestamp: string,
+  metadata: {} | null,
+};
+
+export const ElementTypes = {
+  link: 'link',
+  auth: 'auth',
+  connect: 'connect',
+  balance_transfer: 'balance_transfer',
+};
+
+export type TElementTypes = keyof typeof ElementTypes;
+
+export const ElementProducts = {
```

Comment:
`card` is now `card_brand`, we also allow passing `sensitive` and `payment`

Reply:
good catch